deepset_cloud_sdk/workflows/pipeline_client/pipeline_service.py

```diff
@@ -0,0 +1,257 @@
+        """Append inputs configuration to the YAML string.
+
+        :param yaml_str: Original YAML string
+        :param inputs: Input configuration to append
+        :return: YAML string with inputs appended
+        """
+        input_fields = inputs.model_dump()
+        return self._append_section_to_yaml(yaml_str, "inputs", input_fields)
+
+    def _append_outputs_to_yaml(self, yaml_str: str, outputs: PipelineOutputs) -> str:
+        """Append outputs configuration to the YAML string.
+
+        :param yaml_str: Original YAML string
+        :param outputs: Output configuration to append
+        :return: YAML string with outputs appended
+        """
+        output_fields = outputs.model_dump()
+        return self._append_section_to_yaml(yaml_str, "outputs", output_fields)
+
+    def _create_config_yaml(self, pipeline: PipelineProtocol, config: PublishConfig) -> str:
+        """Create the config YAML string.
+
+        :param pipeline: The Haystack pipeline to convert to YAML
+        :param config: Configuration for publishing
+        :return: Complete YAML string with inputs and outputs
+        """
+        config_yaml = pipeline.dumps()
+        config_yaml = self._append_inputs_to_yaml(config_yaml, config.inputs)
+        config_yaml = self._append_outputs_to_yaml(config_yaml, config.outputs)
+        return config_yaml
+
+    async def _publish_pipeline(self, pipeline: PipelineProtocol, config: PublishConfig) -> None:
+        """Publish a pipeline to deepset Cloud.
+
+        :param pipeline: The pipeline to publish
+        :param config: Configuration for publishing
+        """
+        logger.info(f"Publishing {config.pipeline_type.value}: {config.name}")
+        pipeline_yaml = self._create_config_yaml(pipeline, config)
+        response = await self.api.post(
+            workspace_name=DEFAULT_WORKSPACE_NAME,
+            endpoint="pipelines",
+            json={"name": config.name, "query_yaml": pipeline_yaml},
+        )
+        response.raise_for_status()
+        if response.status_code == 201:
+            logger.info(f"{config.pipeline_type.value} {config.name} successfully created")
+
+    async def _publish_index(self, pipeline: PipelineProtocol, config: PublishConfig) -> None:
+        """Publish an index pipeline to deepset Cloud.
+
+        :param pipeline: The Haystack pipeline to publish
+        :param config: Configuration for publishing
+        """
+        pipeline_yaml = self._create_config_yaml(pipeline, config)
+        response = await self.api.post(
+            workspace_name=DEFAULT_WORKSPACE_NAME,
+            endpoint="indexes",
+            json={"name": config.name, "config_yaml": pipeline_yaml},
+        )
+        response.raise_for_status()
+        if response.status_code == 201:
+            logger.info(f"Index {config.name} successfully created")
+
+    async def publish(self, pipeline: PipelineProtocol, config: PublishConfig) -> None:
```

Comment:
what about returning either the pipeline ID here or an object containing the pipeline ID?

Reply:
Good point, unfortunately our pipeline creation endpoint currently only returns the pipeline name 🙃 and I don't see much value in returning the pipeline name. This is why I skipped it for now. I want to implement the return value in a separate PR. I'm thinking of using the pipeline name, and the link to the pipeline/index in deepset instead.

Reply:
Fair - would be a nice thing to change that in the hub API 👍🏻